compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeSwingLayer.desktop.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:OptIn(ExperimentalComposeUiApi::class)
+
+package androidx.compose.ui.awt
+
+import androidx.compose.ui.ExperimentalComposeUiApi
+import java.awt.Component
+import java.awt.Dimension
+import java.awt.Graphics
+import javax.accessibility.Accessible
+import javax.accessibility.AccessibleContext
+import javax.swing.SwingUtilities
+import org.jetbrains.skiko.ClipRectangle
+import org.jetbrains.skiko.ExperimentalSkikoApi
+import org.jetbrains.skiko.GraphicsApi
+import org.jetbrains.skiko.SkiaLayerAnalytics
+import org.jetbrains.skiko.swing.SkiaSwingLayer
+import org.jetbrains.skiko.swing.SkiaSwingLayerComponent
+
+/**
+ * Provides [component] that can be used as a Swing component.
+ * Content set in [setContent] will be drawn on this [component].
+ *
+ * [ComposeSwingLayer] provides smooth integration with Swing, so z-ordering, double-buffering etc. from Swing will be taken into account.
+ *
+ * However, if smooth interop with Swing is not needed, consider using [androidx.compose.ui.awt.ComposeWindowLayer]
+ */
+internal class ComposeSwingLayer(
+    private val skiaLayerAnalytics: SkiaLayerAnalytics
+) : ComposeLayer<SkiaSwingLayerComponent>() {
+    private val _component = ComposeSwingSkiaLayer()
+    override val component: SkiaSwingLayerComponent get() = _component
+
+    override val renderApi: GraphicsApi
+        get() = _component.renderApi
+
+    override val clipComponents: MutableList<ClipRectangle>
+        get() = _component.clipComponents
+
+    override val focusComponentDelegate: Component
+        get() = _component
+
+    override fun requestNativeFocusOnAccessible(accessible: Accessible) {
+        // TODO: support a11y
+    }
+
+    override fun onComposeInvalidation() {
+        _component.repaint()
+    }
+
+    init {
+        attachComposeToComponent()
+    }
+
+    override fun disposeComponentLayer() {
+        _component.dispose()
+    }
+
+    /**
+     * See also backendLayer for standalone compose [androidx.compose.ui.awt.ComposeWindowLayer.ComposeWindowSkiaLayer]
+     */
+    @OptIn(ExperimentalSkikoApi::class)
+    private inner class ComposeSwingSkiaLayer :
```

Comment:
Code of these layers kinda the same. I thought about having Wrapper JPanel and override these methods there and add SkiaLayer to it, but I think that something may go wrong in this case.